### PR TITLE
JENKINS-64116 Fix compatibility with Jenkins core > 2.263

### PR DESCRIPTION
**Pull request by timja**
https://issues.jenkins.io/browse/JENKINS-64116

Most of the fixes are in azure-commons and requires https://github.com/jenkinsci/azure-commons-plugin/pull/51 to be released first

@xuzhang3

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -32,24 +32,27 @@
...
-        <jenkins.version>2.164.3</jenkins.version>
+        <jenkins.version>2.266</jenkins.version>
         <java.level>8</java.level>
         <findbugs.failOnError>true</findbugs.failOnError>
         <findbugs.excludeFilterFile>findbugs-exclude.xml</findbugs.excludeFilterFile>
         <maven.javadoc.skip>true</maven.javadoc.skip>
         <azure-credentials.version>4.0.2</azure-credentials.version>
         <kubernetes-client.version>2.3.1</kubernetes-client.version>
-        <azure-commons.version>1.0.4</azure-commons.version>
+        <azure-commons.version>1.0.6-rc220.c094849fd287</azure-commons.version>
```
Comment (timja) [inline pom.xml:55]:
```suggestion
        <azure-commons.version>1.1.0</azure-commons.version>
```

File: pom.xml [hunk 2]
```diff
@@ -193,8 +196,8 @@
         <dependencies>
             <dependency>
                 <groupId>io.jenkins.tools.bom</groupId>
-                <artifactId>bom-2.164.x</artifactId>
-                <version>9</version>
+                <artifactId>bom-2.263.x</artifactId>
+                <version>18</version>
```
Comment (timja) [inline pom.xml:200]:
```suggestion
                <version>20</version>
```

**Comment by timja** [comment]
cc @xuzhang3

File: pom.xml [hunk 3]
```diff
@@ -32,24 +32,27 @@
...
 
     <properties>
+        <revision>1.2.1</revision>
+        <changelist>-SNAPSHOT</changelist>
+        <gitHubRepo>jenkinsci/azure-container-agents-plugin</gitHubRepo>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <azuresdk.version>1.19.0</azuresdk.version>
-        <jenkins.version>2.164.3</jenkins.version>
+        <jenkins.version>2.266</jenkins.version>
```
Comment (timja) [inline pom.xml:48]:
```suggestion
        <jenkins.version>2.271</jenkins.version>
```

**Comment by xuzhang3** [comment]
LGTM

**Pull request merged**